libparsec/crates/types/src/sas_code.rs

```diff
@@ -24,16 +24,28 @@ impl std::fmt::Display for SASCode {
     }
 }
 
+#[cfg_attr(test, derive(PartialEq))]
+#[derive(Clone, Debug)]
+pub struct SASCodeParseError(pub(super) String);
```

Comment:
```suggestion
pub struct SASCodeParseError;
```

No need to display the invalid data

Reply:
done